File: BackEndTest/WebApiTest/WebApiTest/Models/ContactModel.cs
using System;

namespace WebApiTest.Models
{
    public class ContactModel : IEquatable<ContactModel>
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string CompanyName { get; set; }

        public bool IsFavorite { get; set; }

        public string SmallImageURL { get; set; }

        public string LargeImageURL { get; set; }

        public string EmailAddress { get; set; }

        public DateTime? Birthdate { get; set; }

        public PhoneModel Phone { get; set; }

        public AddressModel Address { get; set; }

        public bool Equals(ContactModel other)
        {
            return Id == other.Id;
        }
    }
}

File: FrontEndTest/src/models/ContactModel.js
import { AddressModel } from "@/models/AddressModel";
import { PhoneModel } from "@/models/PhoneModel";

export class ContactModel {
    constructor(data) {
        Object.assign(this, data)
        this.phone = Object.assign(new PhoneModel(), this.phone)
        this.address = Object.assign(new AddressModel(), this.address)
    }

    get Id() {
        return this.id
    }

    set Id(value) {
        this.id = value
    }

    get Name() {
        return this.name
    }

    set Name(value) {
        this.name = value
    }

    get CompanyName() {
        return this.companyName
    }

    set CompanyName(value) {
        this.companyName = value
    }


    get IsFavorite() {
        return this.isFavorite
    }

    set IsFavorite(value) {
        this.isFavorite = value
    }


    get SmallImageURL() {
        return this.smallImageURL
    }

    set SmallImageURL(value) {
        this.smallImageURL = value
    }


    get LargeImageURL() {
        return this.largeImageURL
    }

    set LargeImageURL(value) {
        this.largeImageURL = value
    }


    get EmailAddress() {
        return this.emailAddress
    }

    set EmailAddress(value) {
        this.emailAddress = value
    }


    get Birthdate() {
        return this.birthdate
    }

    set Birthdate(value) {
        this.birthdate = value
    }

    get Phone() {
        if (this.phone == null) {
            this.phone = new PhoneModel();
        }

        return this.phone
    }

    set Phone(value) {
        this.phone = Object.assign(new PhoneModel(), value);
    }

    get Address() {
        if (this.address == null) {
            this.address = new AddressModel();
        }

        return this.address
    }

    set Address(value) {
        this.address = Object.assign(new AddressModel(), value);
    }
}